### PR TITLE
Fixes #30265 - Add DSL docs for content facet

### PR DESCRIPTION
**Pull request by ofedoren**
Needs https://github.com/Katello/katello/pull/8801, https://github.com/Katello/katello/pull/8815,  https://github.com/Katello/katello/pull/8816

### CODE REVIEW

**Comment by ares** [comment]
@ofedoren needs rebase now

File: app/models/katello/concerns/content_facet_host_extensions.rb [hunk 1]
```diff
@@ -57,6 +57,13 @@ def content_facet_ignore_update?(attributes)
             attributes['lifecycle_environment_id'].blank?
           )
         end
+
+        apipie :class do
```
Comment (ares) [inline app/models/katello/concerns/content_facet_host_extensions.rb:61]: this now fails with 
```
NoMethodError: undefined method `apipie' for Katello::Concerns::ContentFacetHostExtensions:Module
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.0.0.pre.master/app/models/katello/concerns/content_facet_host_extensions.rb:62:in `<module:ContentFacetHostExtensions>'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.0.0.pre.master/app/models/katello/concerns/content_facet_host_extensions.rb:3:in `<module:Concerns>'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.0.0.pre.master/app/models/katello/concerns/content_facet_host_extensions.rb:2:in `<module:Katello>'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-4.0.0.pre.master/app/models/katello/concerns/content_facet_host_extensions.rb:1:in `<top (required)>'
```

Comment (ofedoren) [reply app/models/katello/concerns/content_facet_host_extensions.rb:61]: that's strange.. I've run `apipie_dsl:cache` and it seems to be working:
ScreenShot-1595420411004

Comment (ares) [reply app/models/katello/concerns/content_facet_host_extensions.rb:61]: FTR adding the `extend ApipieDSL::Class` helped, however Oleh couldn't reproduce, maybe it's something on my side

Comment (jlsherrill) [reply app/models/katello/concerns/content_facet_host_extensions.rb:61]: seems to be working for me, wondering if a 'spring stop' would have solved it?

Comment (ares) [reply app/models/katello/concerns/content_facet_host_extensions.rb:61]: nah, this was prod-like env, anyway, if I was the only one, please ignore

**Comment by jlsherrill** [comment]
Thanks @ofedoren !

**Pull request merged**